File: scripts/upload-to-supabase.js
import { createClient } from '@supabase/supabase-js';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Supabase configuration
const supabaseUrl = 'https://gfbedvoexpulmmfitxje.supabase.co';
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdmYmVkdm9leHB1bG1tZml0eGplIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MjI3OTUzMDAsImV4cCI6MjAzODM3MTMwMH0.r8lp0GGrOFnHgdH3q7S2yNzN5LNGqKalnIsvL1pdM1E';

const supabase = createClient(supabaseUrl, supabaseKey);

// Product image mappings - using existing images as placeholders
const productImages = [
  {
    slug: 'paji-plushie-soft-mascot',
    sourcePath: path.join(__dirname, '../public/images/logos/paji-mascot-full.png'),
    targetName: 'paji-plushie.png',
    bucket: 'products'
  },
  {
    slug: 'paji-squishy-stress-toy',
    sourcePath: path.join(__dirname, '../public/images/logos/paji-mascot-front.png'),
    targetName: 'paji-squishy-toy.png',
    bucket: 'products'
  },
  {
    slug: 'ispeak-teacher-coffee-mug-special',
    sourcePath: path.join(__dirname, '../public/images/logos/ispeak-logo-with-text.png'),
    targetName: 'teacher-coffee-mug.png',
    bucket: 'products'
  },
  {
    slug: 'african-languages-wall-posters',
    sourcePath: path.join(__dirname, '../public/images/logos/ispeak-text-logo.png'),
    targetName: 'wall-posters.png',
    bucket: 'products'
  },
  {
    slug: 'african-tales-storybook-collection',
    sourcePath: path.join(__dirname, '../public/images/logos/paji-mascot-full.png'),
    targetName: 'african-tales-books.png',
    bucket: 'products'
  },
  {
    slug: 'african-heritage-pattern-notebook',
    sourcePath: path.join(__dirname, '../public/images/logos/ispeak-logo-with-text.png'),
    targetName: 'african-pattern-notebook.png',
    bucket: 'products'
  },
  {
    slug: 'yoruba-language-flashcards',
    sourcePath: path.join(__dirname, '../public/images/logos/ispeak-text-logo.png'),
    targetName: 'yoruba-flashcards.png',
    bucket: 'products'
  }
];

async function uploadImage(imageData) {
  try {
    // Check if source file exists
    if (!fs.existsSync(imageData.sourcePath)) {
      console.log(`Source file not found: ${imageData.sourcePath}`);
      return null;
    }

    // Read the file
    const fileBuffer = fs.readFileSync(imageData.sourcePath);
    
    // Upload to Supabase storage
    const { data, error } = await supabase.storage
      .from(imageData.bucket)
      .upload(imageData.targetName, fileBuffer, {
        contentType: 'image/png',
        upsert: true
      });

    if (error) {
      console.error(`Error uploading ${imageData.targetName}:`, error);
      return null;
    }

    // Get public URL
    const { data: urlData } = supabase.storage
      .from(imageData.bucket)
      .getPublicUrl(imageData.targetName);

    console.log(`✓ Uploaded: ${imageData.targetName}`);
    return {
      slug: imageData.slug,
      url: urlData.publicUrl
    };
  } catch (error) {
    console.error(`Failed to upload ${imageData.targetName}:`, error);
    return null;
  }
}

async function updateProductImages() {
  console.log('Starting upload to Supabase...\n');
  
  const uploadedImages = [];
  
  for (const image of productImages) {
    const result = await uploadImage(image);
    if (result) {
      uploadedImages.push(result);
    }
  }
  
  console.log('\nUpload completed!');
  console.log('\nUpdating product records...');
  
  // Update products with new URLs
  for (const img of uploadedImages) {
    const { error } = await supabase
      .from('products')
      .update({ images: [img.url] })
      .eq('slug', img.slug);
      
    if (error) {
      console.error(`Failed to update product ${img.slug}:`, error);
    } else {
      console.log(`✓ Updated product: ${img.slug}`);
    }
  }
  
  console.log('\nAll done!');
}

updateProductImages();

File: ISPEAK_PRODUCT_IMAGES/upload_to_supabase.py
#!/usr/bin/env python3
import os
import sys
from supabase import create_client, Client
import json
from pathlib import Path

# Supabase configuration
SUPABASE_URL = "https://gfbedvoexpulmmfitxje.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdmYmVkdm9leHB1bG1tZml0eGplIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MjgyMTQxNzgsImV4cCI6MjA0Mzc5MDE3OH0.OEo-6jCLYNeBGazF-XnBa1awMd-bxtQfyEamVtIcQGI"

# Create Supabase client
supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

# Product image mapping from catalog
product_images = {
    'language-learning-poster-set': 'language-posters-REAL.jpg',
    'ispeak-alphabet-flashcards': 'alphabet-flashcards-REAL.jpg',
    'african-tales-paji-storybook': 'african-tales-storybook-REAL.jpg',
    'kiswahili-phrases-kids': 'kiswahili-phrases-book-REAL.jpg',
    'ispeak-tshirt-adult': 'ispeak-tshirt-adult.jpg',
    'paji-tshirt-kids': 'paji-tshirt-kids.jpg',
    'ispeak-kids-tshirt': 'ispeak-kids-tshirt.jpg',
    'ispeak-navy-hoodie': 'ispeak-hoodie.jpg',
    'ispeak-cap': 'ispeak-cap.jpg',
    'teacher-coffee-mug': 'teacher-coffee-mug-REAL.jpg',
    'ispeak-tote-bag': 'ispeak-tote-bag.jpg',
    'paji-mascot-backpack': 'ispeak-backpack.jpg',
    'paji-water-bottle': 'water-bottle.jpg',
    'african-heritage-notebook': 'african-patterns-notebook-REAL.jpg',
    'paji-sticker-sheet': 'paji-sticker-sheet.jpg',
    'paji-plush-toy': 'paji-plush-toy-REAL.jpg',
    'paji-squishy-toy': 'paji-squishy-toy.jpg'
}

def upload_image(file_path, storage_path):
    """Upload an image to Supabase storage"""
    try:
        with open(file_path, 'rb') as f:
            file_data = f.read()
            
        # Upload to storage
        response = supabase.storage.from_('ispeak-products').upload(
            path=storage_path,
            file=file_data,
            file_options={"content-type": "image/jpeg", "upsert": "true"}
        )
        
        # Get public URL
        public_url = supabase.storage.from_('ispeak-products').get_public_url(storage_path)
        
        return public_url
    except Exception as e:
        print(f"Error uploading {storage_path}: {str(e)}")
        return None

def main():
    web_optimized_dir = Path('/Users/mac/Downloads/iSPEAK/ISPEAK_PRODUCT_IMAGES/web-optimized')
    thumbnails_dir = web_optimized_dir / 'thumbnails'
    
    uploaded_images = {}
    
    # Upload main images
    print("Uploading main product images...")
    for slug, filename in product_images.items():
        file_path = web_optimized_dir / filename
        if file_path.exists():
            storage_path = f"main/{filename}"
            url = upload_image(file_path, storage_path)
            if url:
                uploaded_images[slug] = {
                    'main': url,
                    'filename': filename
                }
                print(f"✓ Uploaded {filename} -> {storage_path}")
        else:
            # Try without -REAL suffix
            alt_filename = filename.replace('-REAL', '')
            file_path = web_optimized_dir / alt_filename
            if file_path.exists():
                storage_path = f"main/{alt_filename}"
                url = upload_image(file_path, storage_path)
                if url:
                    uploaded_images[slug] = {
                        'main': url,
                        'filename': alt_filename
                    }
                    print(f"✓ Uploaded {alt_filename} -> {storage_path}")
            else:
                print(f"✗ File not found: {filename}")
    
    # Upload thumbnails
    print("\nUploading thumbnail images...")
    for slug, info in uploaded_images.items():
        # Try to find corresponding thumbnail
        base_name = info['filename'].replace('.jpg', '')
        thumb_filename = f"{base_name}_thumb.jpg"
        thumb_path = thumbnails_dir / thumb_filename
        
        if thumb_path.exists():
            storage_path = f"thumbnails/{thumb_filename}"
            url = upload_image(thumb_path, storage_path)
            if url:
                uploaded_images[slug]['thumbnail'] = url
                print(f"✓ Uploaded thumbnail {thumb_filename}")
        else:
            print(f"✗ Thumbnail not found: {thumb_filename}")
    
    # Save URLs to JSON file
    with open('uploaded_urls.json', 'w') as f:
        json.dump(uploaded_images, f, indent=2)
    
    print(f"\n✅ Upload complete! Uploaded {len(uploaded_images)} products")
    print("URLs saved to uploaded_urls.json")
    
    return uploaded_images

if __name__ == "__main__":
    uploaded = main()